### PR TITLE
Make JsonRpc Requesttypes generic

### PR DESCRIPTION
**Pull request by CapCap**
## Motivation
In building the stream RPC there was a need to have similar types, but without being tied to the implementation specific details. This breaks out a piece of #8307 to make reviewing more palatable.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?
yes

## Test Plan
All existing tests should function: this is a generic-ization that should maintain 1:1 functionality with existing implementation

## Related PRs
https://github.com/diem/diem/pull/8307

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Changing JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: json-rpc/src/util.rs [hunk 1]
```diff
@@ -110,8 +116,10 @@ impl SdkInfo {
         let lowercase_user_agent = user_agent.to_lowercase();
         let user_agent_parts: Vec<&str> = lowercase_user_agent.split('/').collect();
         if user_agent_parts.len() == 2 {
-            let language = SdkLang::from_str(&user_agent_parts[0]);
-            let version = SdkVersion::from_str(&user_agent_parts[1]);
+            let language =
+                SdkLang::from_str(&user_agent_parts[0]).expect("could not parse sdk language");
+            let version =
+                SdkVersion::from_str(&user_agent_parts[1]).expect("could not parse sdk version");
```
Comment (xli) [inline json-rpc/src/util.rs:122]: I feel this change makes code worse, as it is confusing whether the expect will cause panic

**Comment by xli** [comment]
I don't understand the need of GenericJsonRpcRequest, the way parse method and method request are different, but the usage is also 2 different code path. What is exactly shared by creating GenericJsonRpcRequest?

File: json-rpc/src/util.rs (hunk 1 above)
Comment (CapCap) [reply json-rpc/src/util.rs:122]: this was forced via clippy wanting `from_str` to use the `FromStr` trait. I'm open to suggestions

Comment (xli) [reply json-rpc/src/util.rs:122]: Rename to SdkLang::parse?

**Pull request closed without merging**